### PR TITLE
Add CI run with new analyzer enabled

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add CI run with new analyzer infrastructure enabled.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -1308,6 +1308,40 @@ jobs:
           docker ps --quiet | xargs --no-run-if-empty docker kill ||:
           docker ps --all --quiet | xargs --no-run-if-empty docker rm -f ||:
           sudo rm -fr "$TEMP_PATH"
+  FunctionalStatelessTestReleaseAnalyzer:
```
Comment (nikitamikhaylov) [inline .github/workflows/pull_request.yml:1311]: Probably at least add another run with sanitizer (ASan)? Or just just replace this Release run with ASan?

Comment (novikd) [reply .github/workflows/pull_request.yml:1311]: I want to start just with one because it requires suppressing errors for some tests.

Comment (tavplubix) [inline .github/workflows/pull_request.yml:1311]: Should we add it to `master.yaml` as well?

File: docker/test/stateless-analyzer/run.sh [hunk 2]
```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
```
Comment (tavplubix) [inline docker/test/stateless-analyzer/run.sh:1]: Let's avoid copy-paste when possible

File: docker/test/util/process_functional_tests_result.py [hunk 3]
```diff
@@ -62,17 +62,25 @@ def process_test_log(log_path):
...
                     test_results.append((test_name, "Timeout", test_time, []))
                 elif FAIL_SIGN in line:
-                    failed += 1
-                    test_results.append((test_name, "FAIL", test_time, []))
+                    if test_name in broken_tests:
+                        success += 1
+                        test_results.append((test_name, "OK", test_time, []))
+                    else:
+                        failed += 1
+                        test_results.append((test_name, "FAIL", test_time, []))
```
Comment (tavplubix) [inline docker/test/util/process_functional_tests_result.py:70]: Consider just skipping broken tests in `clickhouse-test`

**Comment by tavplubix** [comment]
Also, it would be great to add a `no-analyzer` tag just in case, so we will be able to completely disable some tests with analyzer

**Comment by tavplubix** [comment]
> set allow_experimental_analyzer = 0 at the test's beginning

Yes, this will work

**Review by tavplubix — APPROVED** [review]

File: tests/ci/functional_test_check.py [hunk 4]
```diff
@@ -62,6 +64,8 @@ def get_additional_envs(check_name, run_by_hash_num, run_by_hash_total):
 
 
 def get_image_name(check_name):
+    if "analyzer" in check_name.lower():
+        return "clickhouse/stateless-analyzer"
```
Comment (qoega) [inline tests/ci/functional_test_check.py:68]:
```suggestion
        return "clickhouse/stateless-test"
```

**Pull request merged**